addons/website/static/src/components/website_loader/website_loader.js

```diff
@@ -1,29 +1,214 @@
+                        const currentTip = this.waitingTips.find(tip => tip.show);
+                        const nextTip =
+                            this.waitingTips[this.waitingTips.indexOf(currentTip) + 1]
+                            ?? this.waitingTips[0];
+                        currentTip.show = false;
+                        nextTip.show = true;
+                        this.state.title = nextTip.title;
+                    }, 10000);
+
+                    return () => clearInterval(tipsInterval);
+                }
+            },
+            () => [this.waitingTips.length]
+        );
+
+        // Prevent user from closing/refreshing the window
+        useEffect(
+            (isVisible) => {
+                if (isVisible) {
+                    window.addEventListener("beforeunload", this.showRefreshConfirmation);
+                } else {
+                    window.removeEventListener("beforeunload", this.showRefreshConfirmation);
+                }
+
+                return () => window.removeEventListener("beforeunload", this.showRefreshConfirmation);
+            },
+            () => [this.state.isVisible]
+        );
+
         this.props.bus.on('SHOW-WEBSITE-LOADER', this, (props) => {
             this.state.isVisible = true;
             this.state.title = props && props.title;
             this.state.showTips = props && props.showTips;
+            this.state.selectedFeatures = props && props.selectedFeatures;
         });
         this.props.bus.on('HIDE-WEBSITE-LOADER', this, () => {
             for (const key of Object.keys(initialState)) {
                 this.state[key] = initialState[key];
             }
+            clearInterval(tipsInterval);
+            clearInterval(trackModulesInterval);
+        });
+        // Action needed if the app automatically refreshes or redirects the
+        // page without hiding/removing the WebsiteLoader. This should be
+        // called prior to any refresh/redirect if the loader is still visible.
+        this.props.bus.on("PREPARE-OUT-WEBSITE-LOADER", this, () => {
+            window.removeEventListener("beforeunload", this.showRefreshConfirmation);
+            // Most of the time, the server doesn't have time to send back
+            // the last installed modules. This condition just makes sure the
+            // user receives an ultimate feedback which confirms that every
+            // feature has been installed before the page reloads.
+            if (this.featuresInstallInfo.total) {
+                this.featuresInstallInfo.nbInstalled = this.featuresInstallInfo.total;
+            }
         });
     }
+
+    trackModules = async (selectedFeatures) => {
+        const installInfo = await this.rpc(
+            "/website/track_installing_modules",
+            {
+                selected_features: selectedFeatures,
+                total_features: this.featuresInstallInfo.total,
+            },
+            { silent: true });
```

Comment:
It's used so the user doesn't see the spinning wheel when the rpc call is a bit long. (Otherwise, it blocks the page with a spinner until it receives the result.)